app.vue

```diff
@@ -1,21 +1,48 @@
 <script lang="ts" setup>
 import { useAuth0 } from "@auth0/auth0-vue";
 
+const { $appInsights } = useNuxtApp();
+
 const { isLoading, loginWithRedirect, isAuthenticated, error, logout } =
   useAuth0();
 
 watch(
   isLoading,
   async (loading) => {
     if (loading) return;
+
     if (!isAuthenticated.value) {
+      $appInsights.event("auth0 - redirect to login", {});
       await loginWithRedirect();
     }
   },
   { immediate: true },
 );
 
+watch(error, async () => {
+  const e = error.value;
```

Comment:
Can be received as first argument of the function you give as parameter to `watch` - see how we do for `isLoading`.

Reply:
done